ipmininet/iptopo.py

```diff
@@ -58,11 +60,20 @@ def addHost(self, name: str, **kwargs) -> 'HostDescription':
            :param name: the name of the node"""
         return HostDescription(super().addHost(str(name), **kwargs), self)
 
-    def addRouter(self, name: str, **kwargs) -> 'RouterDescription':
+    def addRouter(self,
+                  name: str,
+                  cls: 'Router' = Router,
+                  routerDescription: 'RouterDescription' = RouterDescription,
+                  **kwargs) -> 'RouterDescription':
         """Add a router to the topology
 
-        :param name: the name of the node"""
-        return RouterDescription(self.addNode(str(name), isRouter=True,
+        :param name: the name of the node
+        :param cls: the router class of the node to create (optional)
+        "param routerDescription: the RouterDescription class to return
+            (optional)"""
+        return routerDescription(self.addNode(str(name),
+                                              isRouter=True,
+                                              cls=cls,
```

Comment:
I think it was mainly for convenience, because I was tracing down how `mininet` sets the class with `addNode`. Further, tracing how the `Router` class get's to the node took me some time (at least twice). The redundant optional parameter just provides some documentation.

I can remove it, but I think it will be useful (at least for me the next time I stumble here). Maybe the `OpenR` example https://github.com/cnp3/ipmininet/blob/4d0781ffec97cd7b0da15eee145f66aff4921883/ipmininet/examples/simple_openr_network.py#L25 provides enough documentation though.

Reply:
Btw. since you added typing this is probably a more elegant solution than a `is_router` attribute anyways, I guess ... I really start to appreciate that feature.

Reply:
> Btw. since you added typing this is probably a more elegant solution than a `is_router` attribute anyways, I guess ... I really start to appreciate that feature.

Please forget my last comment. I started refactoring and realized (once again), how the mininet topo/net design works. You are totally right, the default `Node` classes are defined by `IPNet`.

Please have a look at 114fac24c077e7339c60a78b5f88c1232b3ef349. Here I don not introduce another source for default classes while still indicating the `cls` option in the method signature.

Reply:
Instead of ``Union[Type[Router], None] = None``, the handiest notation is ``Optional[Type[Router]]``

Reply:
> Instead of `Union[Type[Router], None] = None`, the handiest notation is `Optional[Type[Router]]`

Nice, good to know. I removed the `cls` param though.